blueocean-dashboard/src/main/js/components/ChangeSet.jsx

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+export default class ChangeSet extends Component {
+
+    componentDidMount() {
+        window.addEventListener('resize', () => this.condense(), true);
+        this.condense();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.changeSet !== this.props.changeSet) {
+            this.condense();
+        }
+    }
+
+    condense() {
+        const domNode = ReactDOM.findDOMNode(this.refs.authorsWrapper); // used to check for overflow
+        const domNodeLink = ReactDOM.findDOMNode(this.refs.authors); // the a which need to refresh the content
+        if (domNode && domNodeLink && domNode.scrollWidth > domNode.clientWidth) {
+            const hint = `${this.props.changeSet.length} changes`;
```

Comment:
why is this setting text content instead of setting state flag indicating it's in condensed mode? :ant: